app/controllers/katello/api/rhsm/candlepin_proxies_controller.rb

```diff
@@ -48,7 +48,8 @@ def repackage_message
     end
 
     rescue_from RestClient::Exception do |e|
-      Rails.logger.error pp_exception(e)
+      Rails.logger.error(pp_exception(e, with_backtrace: false))
+      Rails.logger.error(e.backtrace.select {|line| line.match("katello.*controller")}.first)
```

Comment:
Does this do the same thing as e.backtrace.last (or e.backtrace.first) ?

Reply:
`e.backtrace.last` or `e.backtrace.first` gives the last or first line in the backtrace array. 
`e.backtrace.select {|line| line.match("katello.*controller")}.first` gives the first call from katello controller in the backtrace.

```
/opt/theforeman/tfm/root/usr/share/gems/gems/rest-client-2.0.1/lib/restclient/abstract_response.rb:223:in `exception_with_response'
/opt/theforeman/tfm/root/usr/share/gems/gems/rest-client-2.0.1/lib/restclient/abstract_response.rb:103:in `return!'
/opt/theforeman/tfm/root/usr/share/gems/gems/rest-client-2.0.1/lib/restclient/request.rb:809:in `process_result'
/opt/theforeman/tfm/root/usr/share/gems/gems/rest-client-2.0.1/lib/restclient/request.rb:725:in `block in transmit'
/opt/rh/rh-ruby24/root/usr/share/ruby/net/http.rb:877:in `start'
/opt/theforeman/tfm/root/usr/share/gems/gems/rest-client-2.0.1/lib/restclient/request.rb:715:in `transmit'
/opt/theforeman/tfm/root/usr/share/gems/gems/rest-client-2.0.1/lib/restclient/request.rb:145:in `execute'
/opt/theforeman/tfm/root/usr/share/gems/gems/rest-client-2.0.1/lib/restclient/request.rb:52:in `execute'
/opt/theforeman/tfm/root/usr/share/gems/gems/rest-client-2.0.1/lib/restclient/resource.rb:51:in `get'
/opt/theforeman/tfm/root/usr/share/gems/gems/katello-3.7.0.42/app/lib/katello/http_resource.rb:84:in `get'
/opt/theforeman/tfm/root/usr/share/gems/gems/katello-3.7.0.42/app/lib/katello/resources/candlepin/consumer.rb:20:in `get'
/opt/theforeman/tfm/root/usr/share/gems/gems/katello-3.7.0.42/app/controllers/katello/api/rhsm/candlepin_proxies_controller.rb:312:in `block in find_host'
/usr/share/foreman/app/models/concerns/foreman/thread_session.rb:94:in `as'
/usr/share/foreman/app/models/concerns/foreman/thread_session.rb:100:in `as_anonymous_admin'
/opt/theforeman/tfm/root/usr/share/gems/gems/katello-3.7.0.42/app/controllers/katello/api/rhsm/candlepin_proxies_controller.rb:312:in `find_host'
/opt/theforeman/tfm-ror51/root/usr/share/gems/gems/activesupport-5.1.6/lib/active_support/callbacks.rb:413:in `block in make_lambda'
/opt/theforeman/tfm-ror51/root/usr/share/gems/gems/activesupport-5.1.6/lib/active_support/callbacks.rb:178:in `block (2 levels) in halting_and_conditional'
/opt/theforeman/tfm-ror51/root/usr/share/gems/gems/actionpack-5.1.6/lib/abstract_controller/callbacks.rb:12:in `block (2 levels) in <module:Callbacks>'
/opt/theforeman/tfm-ror51/root/usr/share/gems/gems/activesupport-5.1.6/lib/active_support/callbacks.rb:179:in `block in halting_and_conditional'
/opt/theforeman/tfm-ror51/root/usr/share/gems/gems/activesupport-5.1.6/lib/active_support/callbacks.rb:507:in `block in invoke_before'
/opt/theforeman/tfm-ror51/root/usr/share/gems/gems/activesupport-5.1.6/lib/active_support/callbacks.rb:507:in `each'
/opt/theforeman/tfm-ror51/root/usr/share/gems/gems/activesupport-5.1.6/lib/active_support/callbacks.rb:507:in `invoke_before'
/opt/theforeman/tfm-ror51/root/usr/share/gems/gems/activesupport-5.1.6/lib/active_support/callbacks.rb:106:in `block in run_callbacks'
/opt/theforeman/tfm/root/usr/share/gems/gems/katello-3.7.0.42/app/controllers/katello/api/rhsm/candlepin_proxies_controller.rb:35:in `repackage_message'
...
/usr/share/gems/gems/passenger-4.0.18/lib/phusion_passenger/request_handler/thread_handler.rb:108:in `main_loop'
/usr/share/gems/gems/passenger-4.0.18/lib/phusion_passenger/request_handler.rb:441:in `block (3 levels) in start_threads'
/opt/theforeman/tfm/root/usr/share/gems/gems/logging-2.2.2/lib/logging/diagnostic_context.rb:474:in `block in create_with_logging_context'
2018-12-14T14:43:47 [I|app|d3886]   Rendering text template
2018-12-14T14:43:47 [I|app|d3886]   Rendered text template (0.0ms)
2018-12-14T14:43:47 [I|app|d3886] Completed 410 Gone in 38ms (Views: 0.8ms | ActiveRecord: 10.1ms)
```
In the above case, `e.backtrace.last` or `e.backtrace.first` gives the last or first line in the array which most likely would not be from neither foreman nor katello.
`e.backtrace.select {|line| line.match("katello.*controller")}.first` would select line `/opt/theforeman/tfm/root/usr/share/gems/gems/katello-3.7.0.42/app/controllers/katello/api/rhsm/candlepin_proxies_controller.rb:312:in `block in find_host`

Reply:
@jturel Is there always a katello controller call in the backtrace for these requests?

Reply:
There should be since this method handles requests for methods in this particular controller. Are you asking generally or do you have a suspicion that might not be the case?

Reply:
> There should be since this method handles requests for methods in this particular controller. 

Exactly 👍🏻 

>Are you asking generally or do you have a suspicion that might not be the case?

Just confirm.